<launch>

    <include file="$(find simulation_2017)/launch/RMC_field.launch"/>

    <!--<include file="$(find cr17)/launch/aruco.launch">
      <arg name="rear_camera" default="/camera" />
          <arg name="marker_size" default="0.19685" />
          <arg name="board_directory" default="$(find cr17)/param" />
          <arg name="boards_config" default="$(find cr17)/param/boardsConfiguration.yml" />
          <arg name="intraboard_transform" default="0 -0.231775 0 0 0 0 1 /bigBoard /lilBoard 100" />
          <arg name="world_to_bigBoard_transform" default="0 0 0 0 0.7071067811865475 0.7071067811865475 0 /bigBoard /world 100" />
          <arg name="transform_topic" default="/ar_multi_board/transform" />
      </include>-->

    <!--<param name="magic_modelNamePrefix" value="r2d2" />-->

    <rosparam command="load" file="$(find cr17)/param/beacon_localization.yaml"/>
    <rosparam command="load" file="$(find cr17)/param/topics.yaml"/>

    <node name="navigator" pkg="cr17" type="navigator.py">
        <remap from="cmd_vel" to="/cr17_diff_drive_controller/cmd_vel"/>
    </node>

    <node name="beacon_localization" pkg="cr17" type="beacon_localization.py"/>

    <node name="high_level_state_controller" pkg="cr17" type="high_level_state_controller.py">
        <remap from="/filtered_pose" to="/robot/standard_pose"/>
    </node>

    <node name="path_pub" pkg="cr17" type="path_pub.py"/>

    <node name="magic_odomFromGazebo" pkg="simulation_2017" type="magic_odomFromGazebo.py"/>

</launch>
